spec/features/admin_payroll_runs_spec.rb

```diff
@@ -9,9 +9,9 @@
 
     click_on "Payroll"
 
-    create(:claim, :approved, policy: MathsAndPhysics)
-    create(:claim, :approved, policy: StudentLoans)
-    create(:claim, :approved, policy: StudentLoans)
+    create(:claim, :approved, policy: MathsAndPhysics, national_insurance_number: generate(:national_insurance_number))
```

Comment:
Is there a reason why `national_insurance_number` is being set explicitly here? Normally I see an attribute explicitly set when the value is important to the substance of the spec, but here a) we're setting it to be the same as what the factory would give us anyway, and b) nothing specific in this spec references anything to do with the NI number.

Reply:
So, my thinking was something like this. When we changed the claims factory so that instead of using a constant National Insurance number it generated a new one each time, I worried about how we could assess the impact of this change on all of our existing tests. In the sense of, how could we be sure that the change wasn't affecting the setup of those tests in a way that means that they are no longer testing what their author intended them to test?

So I then wondered a bit about how to write tests that are resilient against changes in the factory, and decided that it would be best, in the places where we want to test what happens when the National Insurance number is either the same or different, to be explicit about this in the test setup.

I'd like to understand how people usually handle this problem of changing factories / fixtures.

Reply:
Just realised something that you can't see from this PR:

> When we changed the claims factory so that instead of using a constant National Insurance number it generated a new one each time

We originally made this change as part of our work on this PR, and then coincidentally you added it in dc6868a too. So the explicit NI numbers originally accompanied our changes to the factory. I guess they look a bit weird in isolation, now I look at it.

Reply:
I think you are spot on in that these specs should ultimately be resilient as much as possible to changes made to the factories. That's why it's important when setting up a record in a spec to be explicit with setting values of attributes that are important to the meat and potatoes of the spec itself.

In this case though, I think we can be pretty confident that the change to the factory was a refinement that made the factory more representative of reality. i.e. that a claim coming in will 99% of the time have a unique NI number, that should be default, and that if you want two claims with the same NI number, _that's when you should be explicit_. I think being explicit with setting a random NI number will mostly be noise, unless we have a strong feeling that the factory will change in the future to make NI number the same by default.

Reply:
That makes a lot of sense, thanks. I'll change it to get rid of the explicit random NI numbers.